clients/hadoopfs/src/main/java/io/lakefs/FileSystemTracer.java

```diff
@@ -0,0 +1,353 @@
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;
+    }
+
+    @Override
+    public boolean mkdirs(Path f, FsPermission permission) throws IOException {
+        LOG.trace("mkdirs(f {}, permission {})", f, permission);
+
+        boolean lakefsRes = lfsFileSystem.mkdirs(f, permission);
+        boolean s3aRes = s3AFileSystem.mkdirs(translateLakeFSPathToS3APath(f), permission);
+        LOG.trace("{}s[mkdirs] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;
+    }
+
+    @Override
+    public FileStatus getFileStatus(Path f) throws IOException {
+        LOG.trace("getFileStatus(f {})", f);
+
+        FileStatus lakefsRes = new FileStatus();
+        FileStatus s3aRes = new FileStatus();
+        Path s3aPath = translateLakeFSPathToS3APath(f);
+        boolean fileNotFoundOnLakeFS = false;
+        boolean fileNotFoundOnLakeS3A = false;
+        try {
+            lakefsRes = lfsFileSystem.getFileStatus(f);
+        } catch (Exception e) {
+            LOG.error("[getFileStatus] Can't get {} file status with lakeFSFileSystem", f);
+            fileNotFoundOnLakeFS = true;
+        }
+        try {
+            s3aRes = s3AFileSystem.getFileStatus(s3aPath);
+        } catch (Exception e) {
+            LOG.error("[getFileStatus] Can't get {} file status with S3AFileSystem", s3aPath);
+            fileNotFoundOnLakeS3A = true;
+        }
+
+        LOG.trace("{}[getFileStatus] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            if (fileNotFoundOnLakeFS) {
+                throw new FileNotFoundException("getFileStatus, not found on lakeFS");
+            }
+            return lakefsRes;
+        }
+
+        if (fileNotFoundOnLakeS3A) {
+            throw new FileNotFoundException("getFileStatus, not found on S3A");
+        }
+        Path lfsPath = translateS3APathToLakeFSPath(s3aPath);
+        s3aRes.setPath(lfsPath);
+        return s3aRes;
+    }
+
+    /**
+     * An output stream encapsulating two output streams, one for each file system the tracer uses.
+     */
+    private class TracerOutputTStream extends OutputStream {
+
+        private FSDataOutputStream lakeFSStream;
+        private FSDataOutputStream s3aStream;
+
+        public TracerOutputTStream(Path f, FsPermission permission, boolean overwrite, int bufferSize,
+                                   short replication, long blockSize, Progressable progress) throws IOException {
+            lakeFSStream = lfsFileSystem.create(f, permission, overwrite, bufferSize, replication, blockSize, progress);
+            s3aStream = s3AFileSystem.create(translateLakeFSPathToS3APath(f), permission, overwrite, bufferSize,
+                    replication, blockSize, progress);
+        }
```

Comment:
The call to `create` should be part of the `create` method.
The stream needs to capture the streams and reflect the operations done on the streams and not how they were created.